flow/connectors/clickhouse/normalize.go

```diff
@@ -524,22 +524,24 @@ func (c *ClickHouseConnector) NormalizeRecords(
 					slog.Int64("endBatchID", endBatchID),
 					slog.Int64("lastNormBatchID", q.lastNormBatchID),
 					slog.Int("parallelWorker", i))
-				if err := c.SetLastNormalizedBatchIDForTable(ctx, req.FlowJobName, q.table, endBatchID); err != nil {
+				if err := c.SetLastNormalizedBatchIDForTable(errCtx, req.FlowJobName, q.table, endBatchID); err != nil {
 					return fmt.Errorf("error while setting last synced batch id for table %s: %w", q.table, err)
 				}
-			}
 
-			c.logger.Info("executed INSERT commands to ClickHouse",
-				slog.Int64("endBatchID", endBatchID),
-				slog.Int64("lastNormBatchID", lastNormBatchID),
-				slog.Int("parallelWorker", i))
+				c.logger.Info("executed INSERT command to ClickHouse",
+					slog.String("table", q.table),
+					slog.Int64("endBatchID", endBatchID),
+					slog.Int64("lastNormBatchID", q.lastNormBatchID),
+					slog.Int("parallelWorker", i))
+			}
 			return nil
 		})
 	}
 
 	for _, tbl := range destinationTableNames {
 		lastNormBatchIDForTable, err := c.GetLastNormalizedBatchIDForTable(ctx, req.FlowJobName, tbl)
 		if err != nil {
+			close(queries)
```

Comment:
This change is good, but holy the existing pattern seems error-prone. One way to make it simpler could be to wrap the whole loop in `func(){}()` with `defer close(queries)` at the beginning